### DISCUSSION TITLE
Seeding (uploading) UX

### DISCUSSION THREAD

**Discussion by GaryElshaw** [transmission/transmission#2417] [Q&A]
I've been wanting to have a discussion about this for a long time because there is often an emphasis, and the complaints i've seen, about Transmission as a download client. Aside from the issues with trackers on MacOS (noted everywhere), I'd like the community to talk about Transmission as a seeding client. I got drunk last week and started a list.

1. Why does the Inspector window disappear when in finder?
2. When seeding, why do peers with 100% appear in the inspector window?
3. 0.0% clients appear in the inspector window. Why do peers that don't require data appear at all? See above.
4. Is there a way to see a peer's status when seeding? Can a seeder see how many others are contributing?
6 Everything feels constantly arbitrary and opaque when uploading. 
5a. Why do peers, who do not require data, appear at all? Depending on the torrent, it is commonplace for Transmission, correctly, but incorrectly, be unable to differentiate that an individual client has chosen to exclude files from the source. Why do they remain if they don't require data? Why do they report to have zero data? How is their privacy or security assured by, fleetingly, revealing their IP address?
5b. Why do peers get 'stuck'? Why are they connected but not receiving data from you as a seeder? Why am I wanting to share, but a peer, seemingly wanting me to share, doesn't receive data? Is it me not sharing, or them unwilling to receive?
5c. Why are the sharing limits how they are? If i can upload at 3MB a second, why is it impossible to focus uploads to an individual, and why are we both opaque?
6. Recently, it appears Transmission takes a preference and excludes less recent peers? Is that what is happening?
5d. Why does a peer want to connect and i seemingly reject the connection?
5e. Why can i, seemingly, only upload to a maximum of 3 peers when i have capacity for more?
5f. Why will a peer connect and receive data, and then stop?
5g. If i am seeding, and no clients appear to connect, why can't i ask to connect to share?

**Comment by ckerr** [comment]
I'm not the best person to answer macOS UI client questions, but can answer some of the behavioral questions

> When seeding, why do peers with 100% appear in the inspector window?

It's possible that you don't know the other peer is a seed until after connecting to it -- e.g. Transmission got the peer info from a tracker or from another peer, and that source may have provided out-of-date state info for the peer. Also, on public torrents it can be desirable to connect to other seeds for the purposes of sharing peer  lists to discover other peers in the swarm.

> Why do peers that don't require data appear at all?

> Why do peers, who do not require data, appear at all?

If you're talking about other seeds, see above answer. If that's not what you're referring to, then I don't understand the question

> How is their privacy or security assured by, fleetingly, revealing their IP address?

How is anyone's privacy or security assured by connecting to peers? I don't understand what this question is getting at.

> Why do peers get 'stuck'?

- Networking issues
- Peer be maxing out its bandwidth elsewhere
- Sometimes a piece is missing from the entire swarm or from a large part of the swarm

> Why are the sharing limits how they are? If i can upload at 3MB a second, why is it impossible to focus uploads to an individual, and why are we both opaque?

Mostly for simplicity's sake. You can increase priority on a single torrent. Does it happen that often that you want to max out transfers to a specific peer but avoid other peers in the same torrent?

> Recently, it appears Transmission takes a preference and excludes less recent peers? Is that what is happening?

I don't understand the question

> Why does a peer want to connect and i seemingly reject the connection?

This needs more detail. Are you using a packet sniffer to watch the traffic? I don't understand what you're referring to wrt Transmission rejecting connections to peers.

> 5g. If i am seeding, and no clients appear to connect, why can't i ask to connect to share?

Transmission is constantly looking for peer candidates to connect to when seeding. What makes you think there are known peers that Transmission is avoiding?
